utils/getTranslation.ts

```diff
@@ -0,0 +1,16 @@
+interface TranslationData {
+  [key: string]: any;
+}
+
+const locales: { [key: string]: () => Promise<TranslationData> } = {
+  en: () => import("../locales/en.json").then((r) => r.default),
+  pt: () => import("../locales/pt.json").then((r) => r.default)
+};
+
+export const getTranslation = (lang: string): Promise<TranslationData> => {
+  if (lang in locales) {
+    return Promise.resolve(locales[lang]());
+  } else {
+    return Promise.resolve(locales.en());
+  }
+};
```

Comment:
we could try other way to avoid `if{}else{}`

```suggestion
export const getTranslation = (lang: string): Promise<TranslationData> => {
  const selectedLang = lang in locales ? locales[lang] : locales.en;
  return Promise.resolve(selectedLang());
};
```
or

```suggestion
export const getTranslation = (lang: string): Promise<TranslationData> => {
  if (lang in locales) {
    return Promise.resolve(locales[lang]());
  } 
  return Promise.resolve(locales.en());  
};
```